modules/sftp-transfer/main.tf

```diff
@@ -0,0 +1,309 @@
+
+  dynamic "endpoint_details" {
+    for_each = local.is_vpc ? [1] : []
+
+    content {
+      subnet_ids             = var.subnet_ids
+      security_group_ids     = local.security_group_enabled ? module.security_group.*.id : var.vpc_security_group_ids
+      vpc_id                 = var.vpc_id
+      address_allocation_ids = var.eip_enabled ? aws_eip.sftp.*.id : var.address_allocation_ids
+    }
+  }
+
+  # The AWS console cheats to set "aws"-prefixed tags which integrates with Route 53
+  # to provision a CNAME for the Transfer Server endpoint, and the documentation at
+  # https://docs.aws.amazon.com/transfer/latest/userguide/requirements-dns.html for
+  # tagging this out-of-band simply doesn't work. Leaving this here to provide more
+  # context for why we have a separately managed DNS RR. -MR
+  # See also: https://github.com/hashicorp/terraform-provider-aws/issues/6956 
+
+  tags = merge(module.this.tags, {
+    "fixme:aws:transfer:route53HostedZoneId" = join("", ["/hostedzone/", local.zone_id])
+    "fixme:aws:transfer:customHostname"      = local.domain_name
+  })
+
+}
+
+resource "aws_transfer_user" "default" {
+  for_each = local.enabled ? var.sftp_users : {}
+
+  server_id = join("", aws_transfer_server.default[*].id)
+  role      = aws_iam_role.s3_access_role.arn
+
+  user_name = each.value.user_name
+
+  home_directory_type = var.restricted_home ? "LOGICAL" : "PATH"
+
+  # https://docs.aws.amazon.com/transfer/latest/userguide/logical-dir-mappings.html#implement-log-dirs
+  dynamic "home_directory_mappings" {
+    for_each = var.restricted_home ? [1] : []
+
+    content {
+      entry  = "/"
+      target = "/${join("", data.aws_s3_bucket.landing[*].id)}/${each.value.user_name}"
+    }
+  }
+
+  tags = module.this.tags
+}
+
+resource "aws_transfer_ssh_key" "default" {
+  for_each = local.enabled ? var.sftp_users : {}
+
+  server_id = join("", aws_transfer_server.default[*].id)
+
+  user_name = each.value.user_name
+  body      = each.value.public_key
+
+  depends_on = [
+    aws_transfer_user.default
+  ]
+}
+
+resource "aws_transfer_user" "admin" {
+  for_each = local.enabled ? var.sftp_admins : {}
+
+  server_id = join("", aws_transfer_server.default[*].id)
+  role      = aws_iam_role.s3_access_role.arn
+
+  user_name = each.value.admin_name
+
+  home_directory_type = "PATH"
+  home_directory      = "/${join("", data.aws_s3_bucket.landing[*].id)}"
+
+  tags = module.this.tags
+}
+
+resource "aws_transfer_ssh_key" "admin" {
+  for_each = local.enabled ? var.sftp_admins : {}
+
+  server_id = join("", aws_transfer_server.default[*].id)
+
+  user_name = each.value.admin_name
+  body      = each.value.public_key
+
+  depends_on = [
+    aws_transfer_user.admin
+  ]
+}
+
+resource "aws_eip" "sftp" {
+  count = local.enabled && var.eip_enabled ? length(var.subnet_ids) : 0
+
+  vpc = local.is_vpc
+}
+
+module "security_group" {
+  source  = "cloudposse/security-group/aws"
+  version = "0.3.1"
```

Comment:
This is an old module and should be updated along with its inputs since 0.3.x was deprecated in favor of 0.4.x